framework/src/preconditioners/FiniteDifferencePreconditioner.C

```diff
@@ -46,11 +46,18 @@ validParams<FiniteDifferencePreconditioner>()
                         "matrix for degrees of freedom that might be coupled "
                         "by inspection of the geometric search objects.");
 
+  MooseEnum finite_difference_type("standard coloring", "coloring");
+  params.addParam<MooseEnum>("finite_difference_type",
+                             finite_difference_type,
+                             "standard: standard finite difference"
+                             "coloring: finite difference based on coloring");
+
   return params;
 }
 
 FiniteDifferencePreconditioner::FiniteDifferencePreconditioner(const InputParameters & params)
-  : MoosePreconditioner(params)
+  : MoosePreconditioner(params),
+    _finite_difference_type(getParam<MooseEnum>("finite_difference_type"))
```

Comment:
Why are we using strings and string comparisons? We built the MooseEnum object to avoid this. If you do string comparisons you have to worry about case too which you aren't doing here. Either use the MooseEnum or create an actual C++ enum to do the comparisons. It's easy to convert the MooseEnum to the C++ enum. We do it all the time.

Reply:
Resolved.